modules/p2/ting_openlist/ting_openlist.module

```diff
@@ -305,62 +257,65 @@ function ting_openlist_get_soap($wsdl = NULL, $trace = FALSE) {
-      );
-
-      // If logging is enabled, log the function name and execution time.
-      if (variable_get('ting_openlist_log', FALSE)) {
-        $execution_time = microtime(TRUE) - $start_time;
-
-        $details = '';
-        if ($detailed_log || variable_get('ting_openlist_log_details', FALSE)) {
-          $details = "<pre>" . var_export($arguments, TRUE) . '</pre>';
-        }
+  timer_start(__FUNCTION__);
 
-        watchdog(
-          'openlist',
-          '@function (@execution_time s)!details',
-          array(
-            '@function' => $function,
-            '@execution_time' => $execution_time,
-            '!details' => $details,
-          ),
-          WATCHDOG_INFO
-        );
-      }
-    }
-    catch (Exception $e) {
-      $execution_time = microtime(TRUE) - $start_time;
-      watchdog(
-        'openlist',
-        'Error (@execution_time s).<br /><strong>Function:</strong> @function.<br /><strong>Exception:</strong>@exception<hr /><pre>!details</pre>',
-        array(
-          '@function' => $function,
-          '@execution_time' => $execution_time,
-          '@exception' => $e->getMessage(),
-          '!details' => var_export($arguments, TRUE),
-        ),
-        WATCHDOG_ERROR
-      );
-
-      throw $e;
-    }
+  $parts = array(
+    variable_get('ting_openlist_domain', TING_OPENLIST_DEFAULT_DOMAIN),
+    'json',
+    $function,
+  );
+
+  $query = array('library_code' => variable_get('ting_library_code', ''));
+
+  if (variable_get('ting_openlist_developer', FALSE)) {
+    $query['developer'] = 'on';
   }
-  else {
-    throw new Exception('No soap client available.');
+
+  $query += $arguments;
+
+  $url = url(
+    implode('/', $parts),
+    array('query' => $query)
+  );
+
+  $response = drupal_http_request($url);
+
+  if ($response->code != '200') {
+    watchdog(
+      'ting_openlist',
+      '<p><strong>Function:</strong><pre>@function</pre></p><p><strong>Arguments:</strong><pre>!arguments</pre></p><p><strong>Response:</strong><pre>!response</pre></p><p><strong>Execution time:</strong><pre>@execution_time</pre></p>',
+      array(
+        '@function' => $function,
+        '@execution_time' => timer_read(__FUNCTION__),
+        '!arguments' => htmlentities(var_export($arguments, TRUE)),
+        '!response' => htmlentities(var_export($response, TRUE)),
+      ),
+      WATCHDOG_ERROR
+    );
+
+    throw new Exception('Ting openlist error.');
+  }
+
+  $data = drupal_json_decode($response->data);
+
+  if (variable_get('ting_openlist_log', FALSE)) {
+    $details = '';
+    if ($detailed_log || variable_get('ting_openlist_log_details', FALSE)) {
+      $details = "<pre>" . htmlentities(var_export($arguments, TRUE)) . '</pre>';
+    }
+    watchdog(
+      'ting_openlist',
+      '@function (@execution_time ms)!details',
+      array(
+        '@function' => $function,
+        '@execution_time' => timer_read(__FUNCTION__),
+        '!details' => $details,
+      ),
+      WATCHDOG_INFO
+    );
```

Comment:
See comment above regarding handling of function/arguments. I would also like to see the response of each request logged [as we do for the Ting Client](https://github.com/ding2/ting-client/blob/master/lib/adapter/TingClientRequestAdapter.php#L72).

With this in mind `ting_openlist_log_details` may be unneeded. Consider refactoring logging to a single function.